src/compas_robots/configuration.py

```diff
@@ -343,25 +343,24 @@ def from_prismatic_and_revolute_values(cls, prismatic_values, revolute_values, j
         )
 
     @property
-    def data(self):
-        """dict : The data representing the configuration.
+    def dtype(self):
+        return "compas_robots/Configuration"
 
-        By assigning a data dictionary to this property, the current data of
-        the configuration will be replaced by the data in the dict. The
-        data getter and setter should always be used in combination with each
-        other.
-        """
+    @property
+    def data(self):
         return {
             "joint_values": self.joint_values,
             "joint_types": self.joint_types,
             "joint_names": self.joint_names,
         }
 
-    @data.setter
-    def data(self, data):
-        self._joint_values = FixedLengthList(data.get("joint_values") or data.get("values") or [])
-        self._joint_types = FixedLengthList(data.get("joint_types") or data.get("types") or [])
-        self._joint_names = FixedLengthList(data.get("joint_names") or [])
+    @classmethod
+    def from_data(cls, data):
+        return cls(
+            data.get("joint_values") or data.get("values") or [],
+            data.get("joint_types") or data.get("types") or [],
+            data.get("joint_names") or [],
```

Comment:
why would there be `values` or `types` keys?

Reply:
Good catch, this is legacy from a very early version of compas_fab. I removed it at https://github.com/compas-dev/compas_robots/pull/1/commits/e6bc719ea9724436cdd11a64f5937255e58f02c4

Reply:
then I guess this `from_data` can also be removed :)